lib/deploy.go

```diff
@@ -15,17 +16,47 @@ import (
 // and then deploys the package to salesforce
 func PushByPaths(fpaths []string, byName bool, namePaths map[string]string, opts *ForceDeployOptions) {
 	pb := NewPushBuilder()
-
 	var badPaths []string
 	for _, fpath := range fpaths {
-		// TODO: check for folder, if a folder, add all files in it
-		name, err := pb.AddFile(fpath)
+
+		fi, err := os.Stat(fpath)
 		if err != nil {
-			fmt.Println(err.Error())
+			fmt.Println(err)
 			badPaths = append(badPaths, fpath)
-		} else {
-			// Store paths by name for error messages
-			namePaths[name] = fpath
+			continue
+		}
+
+		mode := fi.Mode()
+		//If path provided is dir we are adding all containing files to deployment
+		if mode.IsDir() {
```

Comment:
How about refactoring this branch to make use of a new function for adding a directory to the `PackageBuilder`? Maybe:
```
func (pb *PackageBuilder) AddDirectory(fpath string) (namePaths map[string]string, badPaths []string)
````

Reply:
Reasonable, I will update code soon